lib/ansible/modules/storage/netapp/sf_facts.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/python
+# (c) 2017, NetApp, Inc
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+DOCUMENTATION = '''
+module: sf_facts
+
+short_description: Report facts of a NetApp SolidFire cluster.
+extends_documentation_fragment:
+    - netapp.solidfire
+version_added: '2.5'
+author: Kevin Hulquest (@hulquest)
+description:
+- Report the facts of a SolidFire cluster..
+- High level cluster, node, accounts, volume, virtual volume, snapshots, events, cluster faults, and version information, version information.
+
+'''
+
+EXAMPLES = """
+---
+    - name: Get cluster facts.
+      sf_facts:
+        hostname: "{{ solidfire_hostname }}"
+        username: "{{ solidfire_username }}"
+        password: "{{ solidfire_password }}"
+"""
+
+RETURN = """
+msg:
```

Comment:
msg is a common return and must not be documented. But your returns should be identical to the dict of facts returned.